Formula/gnutls.rb

```diff
@@ -77,6 +77,14 @@ def post_install
 
     pkgetc.mkpath
     (pkgetc/"cert.pem").atomic_write(valid_certs.join("\n"))
+
+    # Recompile gnutls.scm since it might have been modified when pouring the
```

Comment:
How long does this take (on your machine)? Could we change the file date on this or another file to avoid this instead?

Reply:
I thought about this initially but then thought that it was not possible.

Please, correct me if wrong, but my understanding from today is that the reason for the @@HOMEBREW_COLLAR@@ placeholder is to be able to install bottles in a different Homebrew location in the install system.

Touching the file would work if both the system that has build the bottle and the system where we are installing had the same Homebrew directories.

Reply:
I'm suggesting maybe touching this file (or others) in `post_install` could work?

Reply:
Yes, I understand. It works as long as the system building the bottle uses `/usr/local` and the system where you install the bottle also uses `/usr/local`.

Reply:
That's what I have, but I have no idea if other people install Homebrew in different locations. If not, then we can just `touch` the file in `post_install` and everything will just work.

Reply:
I'm not sure I understand why it'd be dependent on location if we update timestamps ourselves, can you explain?

Reply:
Absolutely. I'll try to do a better job 😄 .

`gnutls.scm` is auto-generated from a `gnutls.in` during make. It replaces (using sed) a placeholder and the replacement ends up being `/usr/local/Cellar/gnutls/3.6.14/lib/guile/3.0/extensions`. This is fine. Then, `make` compiles gnutls.scm into gnutls.go. So far so good.

From my investigations today I learned that when a bottle is built, all text files (except some Metadata files) are searched replacing Homebrew paths for placeholders. gnutls.scm is a text file which means `/usr/local/Cellar/gnutls/3.6.14/lib/guile/3.0/extensions` gets replaced for `@@HOMEBREW_CELLAR@@/3.6.14/lib/guile/3.0/extensions` (this will be in the bottle).

Now, when installing the bottle (after pouring) the placeholder @@HOMEBREW_CELLAR@@ will get replaced by Hombrew Cellar prefix, which could differ in the system where we are installing.

At this point we have `gnutls.scm` with a newer time (since the placeholder has just been replaced) than `gnutls.go`. Now, when you try to use the Gnutls bindings in Guile, Guile will say "hey! gnutls.go has an older modification time that gnutls.scm" and it will recompile it. This is fine, but ideally it should not recompile.

So, touching `gnutls.go` will work if the system that builds the bottle has a Homebrew prefix of `/usr/local` and the system where we are installing the bottle also has a Homebrew prefix of `/usr/local`.

But imagine if my Hombrew prefix is `/opt`, now we would have `gnutls.scm` pointing to `/opt/Cellar/gnutls/3.6.14/lib/guile/3.0/extension` (because of the @@HOMEBREW_CELLAR@@ placeholder replacement) but `gnutls.go` would still be the compiled file on the system that build the bottle which had `/usr/local/Cellar/gnutls/3.6.14/lib/guile/3.0/extension`.

Reply:
> But imagine if my Hombrew prefix is `/opt`, now we would have `gnutls.scm` pointing to `/opt/Cellar/gnutls/3.6.14/lib/guile/3.0/extension` (because of the @@HOMEBREW_CELLAR@@ placeholder replacement) but `gnutls.go` would still be the compiled file on the system that built the bottle which had `/usr/local/Cellar/gnutls/3.6.14/lib/guile/3.0/extension`.

Ah, I see. In this case: the bottle is not `cellar :any` so it will only be used in `/usr/local` installations because it cannot be relocated so: the proposed touching seems like it should be fine 👍🏻

Reply:
Had no idea about that. Sorry, this is all new to me. I will try this right away.

Any thoughts on making it `depends_on "guile" => :build` so users don't need to install `guile` and adding a caveat to `brew install guile`? I assume this would work right? I mean, bindings would still be built and installed but they would not work unless you `brew install guile`.

Reply:
Confirmed: touching the file works. Just pushed the change. Thank you so much for taking the time to read my messages.

Let me know about `depends_on` and I'll use this same PR to make the update.

Reply:
> Any thoughts on making it `depends_on "guile" => :build` so users don't need to install `guile` and adding a caveat to `brew install guile`? I assume this would work right? I mean, bindings would still be built and installed but they would not work unless you `brew install guile`.



> Confirmed: touching the file works. Just pushed the change. Thank you so much for taking the time to read my messages.

Both sound good!

Reply:
Actually to agree with @smillerdev below: I think you can omit the caveat and just change the dependency and add the touch.